lib/cabin/rspec/cabin_helper.rb

```diff
@@ -0,0 +1,106 @@
+module CabinHelper
+  class LoggingReceiver
+    def filter(&block)
+      cache.detect(&block)
+    end
+
+    def cache
+      @cache ||= []
+    end
+
+    def <<(hash)
+      cache.push(hash)
+    end
+
+    def reset
+      cache.clear
+    end
+  end
+
+  class ReceiveLogMessage
+    def initialize(receiver, message, query)
+      @receiver = receiver
+      @message = message
+      @query = query
+      @compound = false
+      prepare_query
+    end
+
+    def matches?(actual)
+      filterable = actual.is_a?(LoggingReceiver) ? actual : @receiver
+      actual.call if actual.is_a?(Proc)
+      filterable.filter do |h|
+        result = if @compound
+            inter = @transform.call(h[@key])
+            !!(inter.send(@operator, @rhs))
+          else
+            true
+          end
+        result && h[:message].to_s == @message
+      end
+    end
+
+    def failure_message
+      "expected a matching log entry"
+    end
+
+    def failure_message_when_negated
+      "matching log entry was not expected"
+    end
+
+    def supports_block_expectations?
+      true
+    end
+
+    def diffable?
+      false
+    end
+
+    private
+
+    def prepare_query
+      return if !query_valid?
+      @compound = true
+      @operator = :==
+      @transform = ->(v){v}
+      @key = @query[:key].to_sym
+      @rhs = @query[:match]
+      if @rhs.is_a?(Regexp)
+        @operator = :=~
+        @transform = ->(v){v.inspect}
```

Comment:
Using `->` makes this code not work in Ruby 1.8 (I think), and cabin is still used heavily by things that support Ruby 1.8 (fpm). Can you switch to lambda here?

The alternative is for me to choose to drop Ruby 1.8 support for FPM, which I'm not sure I'm ready to do yet.